packages/server/lib/controllers/connect/postSessions.ts

```diff
@@ -0,0 +1,121 @@
+                            organizationId: z.string().max(255).min(0),
+                            displayName: z.string().max(255).optional()
+                        })
+                        .optional()
+                }),
+                allowedIntegrations: z.array(z.string()).optional(),
+                integrationsConfigDefaults: z
+                    .record(
+                        z.object({
+                            connectionConfig: z.record(z.unknown())
+                        })
+                    )
+                    .optional()
+            })
+            .strict()
+            .parse(data)
+});
+
+const handler = asyncWrapper<PostConnectSessions>(async (req, res) => {
+    await db.knex.transaction(async (trx) => {
+        // Check if the linkedProfile exists in the database
+        const getLinkedProfile = await linkedProfileService.getLinkedProfile(trx, {
+            profileId: req.body.linkedProfile.profileId,
+            accountId: res.locals.account.id,
+            environmentId: res.locals.environment.id
+        });
+
+        let linkedProfileId: number;
+        if (getLinkedProfile.isErr()) {
+            // create linked profile if it doesn't exist yet
+            if (getLinkedProfile.error.code === 'not_found') {
+                // fail if linkedProfile doesn't exist and email is not provided
+                if (!req.body.linkedProfile.email) {
+                    res.status(400).send({
+                        error: {
+                            code: 'invalid_request',
+                            errors: [{ code: 'invalid_type', message: 'email is required', path: ['linkedProfile', 'email'] }]
+                        }
+                    });
+                    return;
+                }
+
+                const createLinkedProfile = await linkedProfileService.createLinkedProfile(trx, {
+                    profileId: req.body.linkedProfile.profileId,
+                    email: req.body.linkedProfile.email,
+                    displayName: req.body.linkedProfile.displayName || null,
+                    organization: req.body.linkedProfile.organization?.organizationId
+                        ? {
+                              organizationId: req.body.linkedProfile.organization.organizationId,
+                              displayName: req.body.linkedProfile.organization.displayName || null
+                          }
+                        : null,
+                    accountId: res.locals.account.id,
+                    environmentId: res.locals.environment.id
+                });
+                if (createLinkedProfile.isErr()) {
+                    res.status(500).send({ error: { code: 'internal_error', message: 'Failed to create linked profile' } });
+                    return;
+                }
+                linkedProfileId = createLinkedProfile.value.id;
+            } else {
+                console.log(getLinkedProfile.error);
```

Comment:
if you want to keep it
```suggestion
                logger.error(getLinkedProfile.error);
```

Reply:
🙈